### ISSUE TITLE
Memfs FileInfo IsDir() inconsistent with Mode().IsDir()

### ISSUE THREAD

**Issue by ashb** [spf13/afero#108]
(I'm just recording all the small inconsistencies I find as I use this project, otherwise it's great!:

```go
vfs := afero.NewMemMapFs()
vfs.Mkdir("/foo", 0600)
fi, _ := vfs.Stat("/foo")
if fi.IsDir() != fi.Mode().IsDir() {
    panic("Inconsistent!")
}
```

**Comment by DarcInc** [comment]
Yep, just hit the same thing.

**Comment by mbertschler** [comment]
Fixed by #136
